python/src/indices.rs

```diff
@@ -269,12 +287,97 @@ pub fn shuffle_transformed_vectors(
     }
 }
 
+async fn do_load_shuffled_vectors(
+    filenames: Vec<String>,
+    dir_path: &str,
+    dataset: &Dataset,
+    column: &str,
+    mut ivf_model: IvfModel,
+    pq_model: ProductQuantizer,
+) -> PyResult<()> {
+    let (_, path) = object_store_from_uri_or_path(dir_path).await?;
+    let streams = load_partitioned_shuffles(path.clone(), filenames)
+        .await
+        .infer_error()?;
+
+    let obj_store = dataset.ds.object_store();
+    let path = dataset.ds.indices_dir().child("shuffled_vectors.idx");
+    let mut writer = obj_store.create(&path).await.infer_error()?;
+    write_pq_partitions(&mut writer, &mut ivf_model, Some(streams), None)
+        .await
+        .infer_error()?;
+    let index_name = "ivf_pq_index";
```

Comment:
Let's stick with the existing pattern and use `{column_name}_idx`.  Also, this should be an optional input parameter.